include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -348,25 +400,50 @@ struct ModelMetadata : Base<OrtModelMetadata> {
   int64_t GetVersion() const;
 };
 
+/** \brief Wrapper around ::OrtSession
+*
+*/
 struct Session : Base<OrtSession> {
-  explicit Session(std::nullptr_t) {}
+  explicit Session(std::nullptr_t) {} ///< \brief Create an empty Session object, must be assigned a valid one to be used 
   Session(Env& env, const ORTCHAR_T* model_path, const SessionOptions& options);
   Session(Env& env, const ORTCHAR_T* model_path, const SessionOptions& options, OrtPrepackedWeightsContainer* prepacked_weights_container);
   Session(Env& env, const void* model_data, size_t model_data_length, const SessionOptions& options);
 
-  // Run that will allocate the output values
+  /** \brief Run the model returning results in an Ort allocated vector.
+  * 
+  * The caller provides a list of inputs and a list of the desired outputs to return.
+  *
+  * See the output logs for more information on warnings/errors that occur while processing the model.
+  * Common errors are.. (TODO)
+  * 
+  * \param[in] run_options
+  * \param[in] input_names Array of null terminated strings of length input_count that is the list of input names
```

Comment:
>null terminated [](http://example.com/codeflow?start=36&length=15)

is there a distinction between "null terminated" and "C style" strings?

Reply:
I really hope not. But we need to make it clear about the UTF-8 strings (though not sure where that actually matters yet). That'll come later.

Reply:
why are input_names "null terminated strings" and output_names "C style strings"?